// @shopify/react-i18n/build/cjs/constants/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

exports.DateStyle = void 0;

(function (DateStyle) {
  DateStyle["Long"] = "Long";
  DateStyle["Short"] = "Short";
  DateStyle["Humanize"] = "Humanize";
  DateStyle["Time"] = "Time";
})(exports.DateStyle || (exports.DateStyle = {}));

const dateStyle = {
  [exports.DateStyle.Long]: {
    weekday: 'long',
    month: 'long',
    day: '2-digit',
    year: 'numeric'
  },
  [exports.DateStyle.Short]: {
    month: 'short',
    day: 'numeric',
    year: 'numeric'
  },
  [exports.DateStyle.Humanize]: {
    month: 'long',
    day: 'numeric',
    year: 'numeric'
  },
  [exports.DateStyle.Time]: {
    hour: '2-digit',
    minute: '2-digit'
  }
};
exports.Weekday = void 0;

(function (Weekday) {
  Weekday["Sunday"] = "sunday";
  Weekday["Monday"] = "monday";
  Weekday["Tuesday"] = "tuesday";
  Weekday["Wednesday"] = "wednesday";
  Weekday["Thursday"] = "thursday";
  Weekday["Friday"] = "friday";
  Weekday["Saturday"] = "saturday";
})(exports.Weekday || (exports.Weekday = {}));

const DEFAULT_WEEK_START_DAY = exports.Weekday.Sunday;
const WEEK_START_DAYS = new Map([// Saturday
['AE', exports.Weekday.Saturday], ['AF', exports.Weekday.Saturday], ['BH', exports.Weekday.Saturday], ['DZ', exports.Weekday.Saturday], ['EG', exports.Weekday.Saturday], ['IQ', exports.Weekday.Saturday], ['IR', exports.Weekday.Saturday], ['JO', exports.Weekday.Saturday], ['KW', exports.Weekday.Saturday], ['LY', exports.Weekday.Saturday], ['OM', exports.Weekday.Saturday], ['QA', exports.Weekday.Saturday], ['SA', exports.Weekday.Saturday], ['SY', exports.Weekday.Saturday], ['YE', exports.Weekday.Saturday], // Sunday
['AR', exports.Weekday.Sunday], ['BO', exports.Weekday.Sunday], ['BR', exports.Weekday.Sunday], ['BZ', exports.Weekday.Sunday], ['CA', exports.Weekday.Sunday], ['CL', exports.Weekday.Sunday], ['CO', exports.Weekday.Sunday], ['CR', exports.Weekday.Sunday], ['DO', exports.Weekday.Sunday], ['EC', exports.Weekday.Sunday], ['GT', exports.Weekday.Sunday], ['HK', exports.Weekday.Sunday], ['HN', exports.Weekday.Sunday], ['IL', exports.Weekday.Sunday], ['JM', exports.Weekday.Sunday], ['JP', exports.Weekday.Sunday], ['KE', exports.Weekday.Sunday], ['KR', exports.Weekday.Sunday], ['MO', exports.Weekday.Sunday], ['MX', exports.Weekday.Sunday], ['NI', exports.Weekday.Sunday], ['PA', exports.Weekday.Sunday], ['PE', exports.Weekday.Sunday], ['PH', exports.Weekday.Sunday], ['SG', exports.Weekday.Sunday], ['SV', exports.Weekday.Sunday], ['TW', exports.Weekday.Sunday], ['US', exports.Weekday.Sunday], ['VE', exports.Weekday.Sunday], ['ZA', exports.Weekday.Sunday], ['ZW', exports.Weekday.Sunday], // Monday
['AD', exports.Weekday.Monday], ['AL', exports.Weekday.Monday], ['AM', exports.Weekday.Monday], ['AU', exports.Weekday.Monday], ['AZ', exports.Weekday.Monday], ['BE', exports.Weekday.Monday], ['BG', exports.Weekday.Monday], ['BN', exports.Weekday.Monday], ['BY', exports.Weekday.Monday], ['CH', exports.Weekday.Monday], ['CN', exports.Weekday.Monday], ['CZ', exports.Weekday.Monday], ['DE', exports.Weekday.Monday], ['DK', exports.Weekday.Monday], ['EE', exports.Weekday.Monday], ['ES', exports.Weekday.Monday], ['FI', exports.Weekday.Monday], ['FR', exports.Weekday.Monday], ['GB', exports.Weekday.Monday], ['GE', exports.Weekday.Monday], ['GF', exports.Weekday.Monday], ['GR', exports.Weekday.Monday], ['HR', exports.Weekday.Monday], ['HU', exports.Weekday.Monday], ['ID', exports.Weekday.Monday], ['IE', exports.Weekday.Monday], ['IN', exports.Weekday.Monday], ['IS', exports.Weekday.Monday], ['IT', exports.Weekday.Monday], ['KG', exports.Weekday.Monday], ['KZ', exports.Weekday.Monday], ['LB', exports.Weekday.Monday], ['LT', exports.Weekday.Monday], ['LU', exports.Weekday.Monday], ['LV', exports.Weekday.Monday], ['MA', exports.Weekday.Monday], ['MC', exports.Weekday.Monday], ['MK', exports.Weekday.Monday], ['MN', exports.Weekday.Monday], ['MY', exports.Weekday.Monday], ['NL', exports.Weekday.Monday], ['NO', exports.Weekday.Monday], ['NZ', exports.Weekday.Monday], ['PK', exports.Weekday.Monday], ['PL', exports.Weekday.Monday], ['PT', exports.Weekday.Monday], ['PY', exports.Weekday.Monday], ['RO', exports.Weekday.Monday], ['RS', exports.Weekday.Monday], ['RU', exports.Weekday.Monday], ['SE', exports.Weekday.Monday], ['SK', exports.Weekday.Monday], ['TH', exports.Weekday.Monday], ['TN', exports.Weekday.Monday], ['TR', exports.Weekday.Monday], ['UA', exports.Weekday.Monday], ['UY', exports.Weekday.Monday], ['UZ', exports.Weekday.Monday], ['VN', exports.Weekday.Monday], ['XK', exports.Weekday.Monday]]);
/* eslint-disable line-comment-position */
// See https://en.wikipedia.org/wiki/Right-to-left

const RTL_LANGUAGES = ['ae', // Avestan
'ar', // 'العربية', Arabic
'arc', // Aramaic
'bcc', // 'بلوچی مکرانی', Southern Balochi
'bqi', // 'بختياري', Bakthiari
'ckb', // 'Soranî / کوردی', Sorani
'dv', // Dhivehi
'fa', // 'فارسی', Persian
'glk', // 'گیلکی', Gilaki
'he', // 'עברית', Hebrew
'ku', // 'Kurdî / كوردی', Kurdish
'mzn', // 'مازِرونی', Mazanderani
'nqo', // N'Ko
'pnb', // 'پنجابی', Western Punjabi
'ps', // 'پښتو', Pashto,
'sd', // 'سنڌي', Sindhi
'ug', // 'Uyghurche / ئۇيغۇرچە', Uyghur
'ur', // 'اردو', Urdu
'yi' // 'ייִדיש', Yiddish
];
const EASTERN_NAME_ORDER_FORMATTERS = new Map([['ko', (firstName, lastName, full) => full ? `${lastName}${firstName}` : lastName], ['ja', (firstName, lastName, full) => full ? `${lastName}${firstName}` : `${lastName}様`], ['zh-CN', (firstName, lastName, full) => full ? `${lastName}${firstName}` : lastName], ['zh-TW', (firstName, lastName, full) => full ? `${lastName}${firstName}` : lastName]]);

exports.DEFAULT_WEEK_START_DAY = DEFAULT_WEEK_START_DAY;
exports.EASTERN_NAME_ORDER_FORMATTERS = EASTERN_NAME_ORDER_FORMATTERS;
exports.RTL_LANGUAGES = RTL_LANGUAGES;
exports.WEEK_START_DAYS = WEEK_START_DAYS;
exports.dateStyle = dateStyle;
