// @shopify/ui-extensions-react/build/esm/surfaces/admin.mjs
export { extend, extension } from '@shopify/ui-extensions/admin';
export { reactExtension, reactExtension as render } from './admin/render.mjs';
export { AdminAction } from './admin/components/AdminAction/AdminAction.mjs';
export { AdminBlock } from './admin/components/AdminBlock/AdminBlock.mjs';
export { BlockStack } from './admin/components/BlockStack/BlockStack.mjs';
export { Box } from './admin/components/Box/Box.mjs';
export { Button } from './admin/components/Button/Button.mjs';
export { Checkbox } from './admin/components/Checkbox/Checkbox.mjs';
export { CustomerSegmentationTemplate } from './admin/components/CustomerSegmentationTemplate/CustomerSegmentationTemplate.mjs';
export { Divider } from './admin/components/Divider/Divider.mjs';
export { EmailField } from './admin/components/EmailField/EmailField.mjs';
export { Form } from './admin/components/Form/Form.mjs';
export { Heading } from './admin/components/Heading/Heading.mjs';
export { HeadingGroup } from './admin/components/HeadingGroup/HeadingGroup.mjs';
export { Icon } from './admin/components/Icon/Icon.mjs';
export { Image } from './admin/components/Image/Image.mjs';
export { InlineStack } from './admin/components/InlineStack/InlineStack.mjs';
export { Link } from './admin/components/Link/Link.mjs';
export { NumberField } from './admin/components/NumberField/NumberField.mjs';
export { PasswordField } from './admin/components/PasswordField/PasswordField.mjs';
export { Pressable } from './admin/components/Pressable/Pressable.mjs';
export { ResourceItem } from './admin/components/ResourceItem/ResourceItem.mjs';
export { ResourceList } from './admin/components/ResourceList/ResourceList.mjs';
export { Select } from './admin/components/Select/Select.mjs';
export { Text } from './admin/components/Text/Text.mjs';
export { TextArea } from './admin/components/TextArea/TextArea.mjs';
export { TextField } from './admin/components/TextField/TextField.mjs';
export { Thumbnail } from './admin/components/Thumbnail/Thumbnail.mjs';
export { URLField } from './admin/components/URLField/URLField.mjs';
export { useApi, useExtensionApi } from './admin/hooks/api.mjs';
