// @shopify/discount-app-components/build/cjs/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

exports.Weekday = void 0;

(function (Weekday) {
  Weekday[Weekday["Sunday"] = 0] = "Sunday";
  Weekday[Weekday["Monday"] = 1] = "Monday";
  Weekday[Weekday["Tuesday"] = 2] = "Tuesday";
  Weekday[Weekday["Wednesday"] = 3] = "Wednesday";
  Weekday[Weekday["Thursday"] = 4] = "Thursday";
  Weekday[Weekday["Friday"] = 5] = "Friday";
  Weekday[Weekday["Saturday"] = 6] = "Saturday";
})(exports.Weekday || (exports.Weekday = {}));

const DEFAULT_WEEK_START_DAY = exports.Weekday.Sunday;
exports.DiscountMethod = void 0;

(function (DiscountMethod) {
  DiscountMethod["Code"] = "Code";
  DiscountMethod["Automatic"] = "Automatic";
})(exports.DiscountMethod || (exports.DiscountMethod = {}));

exports.DiscountClass = void 0;

(function (DiscountClass) {
  DiscountClass["Product"] = "PRODUCT";
  DiscountClass["Order"] = "ORDER";
  DiscountClass["Shipping"] = "SHIPPING";
})(exports.DiscountClass || (exports.DiscountClass = {}));

const DEFAULT_DISCOUNT_CODE_LENGTH = 12;
exports.RecurringPaymentType = void 0;

(function (RecurringPaymentType) {
  RecurringPaymentType["FirstPayment"] = "FIRST_PAYMENT";
  RecurringPaymentType["MultiplePayments"] = "MULTIPLE_PAYMENTS";
  RecurringPaymentType["AllPayments"] = "ALL_PAYMENTS";
})(exports.RecurringPaymentType || (exports.RecurringPaymentType = {}));

exports.RequirementType = void 0;

(function (RequirementType) {
  RequirementType["None"] = "NONE";
  RequirementType["Subtotal"] = "SUBTOTAL";
  RequirementType["Quantity"] = "QUANTITY";
})(exports.RequirementType || (exports.RequirementType = {}));

exports.AppliesTo = void 0;

(function (AppliesTo) {
  AppliesTo["Order"] = "ORDER";
  AppliesTo["Products"] = "PRODUCTS";
  AppliesTo["Collections"] = "COLLECTIONS";
})(exports.AppliesTo || (exports.AppliesTo = {}));

exports.CountrySelectionType = void 0;

(function (CountrySelectionType) {
  CountrySelectionType["AllCountries"] = "ALL_COUNTRIES";
  CountrySelectionType["SelectedCountries"] = "SELECTED_COUNTRIES";
})(exports.CountrySelectionType || (exports.CountrySelectionType = {}));

exports.DiscountStatus = void 0;

(function (DiscountStatus) {
  DiscountStatus["Active"] = "ACTIVE";
  DiscountStatus["Expired"] = "EXPIRED";
  DiscountStatus["Scheduled"] = "SCHEDULED";
})(exports.DiscountStatus || (exports.DiscountStatus = {}));

exports.PurchaseType = void 0;

(function (PurchaseType) {
  PurchaseType["OneTimePurchase"] = "ONE_TIME_PURCHASE";
  PurchaseType["Subscription"] = "SUBSCRIPTION";
  PurchaseType["Both"] = "BOTH";
})(exports.PurchaseType || (exports.PurchaseType = {}));

exports.Eligibility = void 0;

(function (Eligibility) {
  Eligibility["Everyone"] = "EVERYONE";
  Eligibility["Customers"] = "CUSTOMERS";
  Eligibility["CustomerSegments"] = "CUSTOMER_SEGMENTS";
})(exports.Eligibility || (exports.Eligibility = {}));

exports.SupportedCountryCode = void 0;

(function (SupportedCountryCode) {
  SupportedCountryCode["Af"] = "AF";
  SupportedCountryCode["Ax"] = "AX";
  SupportedCountryCode["Al"] = "AL";
  SupportedCountryCode["Dz"] = "DZ";
  SupportedCountryCode["Ad"] = "AD";
  SupportedCountryCode["Ao"] = "AO";
  SupportedCountryCode["Ai"] = "AI";
  SupportedCountryCode["Ag"] = "AG";
  SupportedCountryCode["Ar"] = "AR";
  SupportedCountryCode["Am"] = "AM";
  SupportedCountryCode["Aw"] = "AW";
  SupportedCountryCode["Ac"] = "AC";
  SupportedCountryCode["Au"] = "AU";
  SupportedCountryCode["At"] = "AT";
  SupportedCountryCode["Az"] = "AZ";
  SupportedCountryCode["Bs"] = "BS";
  SupportedCountryCode["Bh"] = "BH";
  SupportedCountryCode["Bd"] = "BD";
  SupportedCountryCode["Bb"] = "BB";
  SupportedCountryCode["By"] = "BY";
  SupportedCountryCode["Be"] = "BE";
  SupportedCountryCode["Bz"] = "BZ";
  SupportedCountryCode["Bj"] = "BJ";
  SupportedCountryCode["Bm"] = "BM";
  SupportedCountryCode["Bt"] = "BT";
  SupportedCountryCode["Bo"] = "BO";
  SupportedCountryCode["Ba"] = "BA";
  SupportedCountryCode["Bw"] = "BW";
  SupportedCountryCode["Bv"] = "BV";
  SupportedCountryCode["Br"] = "BR";
  SupportedCountryCode["Io"] = "IO";
  SupportedCountryCode["Bn"] = "BN";
  SupportedCountryCode["Bg"] = "BG";
  SupportedCountryCode["Bf"] = "BF";
  SupportedCountryCode["Bi"] = "BI";
  SupportedCountryCode["Kh"] = "KH";
  SupportedCountryCode["Ca"] = "CA";
  SupportedCountryCode["Cv"] = "CV";
  SupportedCountryCode["Bq"] = "BQ";
  SupportedCountryCode["Ky"] = "KY";
  SupportedCountryCode["Cf"] = "CF";
  SupportedCountryCode["Td"] = "TD";
  SupportedCountryCode["Cl"] = "CL";
  SupportedCountryCode["Cn"] = "CN";
  SupportedCountryCode["Cx"] = "CX";
  SupportedCountryCode["Cc"] = "CC";
  SupportedCountryCode["Co"] = "CO";
  SupportedCountryCode["Km"] = "KM";
  SupportedCountryCode["Cg"] = "CG";
  SupportedCountryCode["Cd"] = "CD";
  SupportedCountryCode["Ck"] = "CK";
  SupportedCountryCode["Cr"] = "CR";
  SupportedCountryCode["Hr"] = "HR";
  SupportedCountryCode["Cu"] = "CU";
  SupportedCountryCode["Cw"] = "CW";
  SupportedCountryCode["Cy"] = "CY";
  SupportedCountryCode["Cz"] = "CZ";
  SupportedCountryCode["Ci"] = "CI";
  SupportedCountryCode["Dk"] = "DK";
  SupportedCountryCode["Dj"] = "DJ";
  SupportedCountryCode["Dm"] = "DM";
  SupportedCountryCode["Do"] = "DO";
  SupportedCountryCode["Ec"] = "EC";
  SupportedCountryCode["Eg"] = "EG";
  SupportedCountryCode["Sv"] = "SV";
  SupportedCountryCode["Gq"] = "GQ";
  SupportedCountryCode["Er"] = "ER";
  SupportedCountryCode["Ee"] = "EE";
  SupportedCountryCode["Sz"] = "SZ";
  SupportedCountryCode["Et"] = "ET";
  SupportedCountryCode["Fk"] = "FK";
  SupportedCountryCode["Fo"] = "FO";
  SupportedCountryCode["Fj"] = "FJ";
  SupportedCountryCode["Fi"] = "FI";
  SupportedCountryCode["Fr"] = "FR";
  SupportedCountryCode["Gf"] = "GF";
  SupportedCountryCode["Pf"] = "PF";
  SupportedCountryCode["Tf"] = "TF";
  SupportedCountryCode["Ga"] = "GA";
  SupportedCountryCode["Gm"] = "GM";
  SupportedCountryCode["Ge"] = "GE";
  SupportedCountryCode["De"] = "DE";
  SupportedCountryCode["Gh"] = "GH";
  SupportedCountryCode["Gi"] = "GI";
  SupportedCountryCode["Gr"] = "GR";
  SupportedCountryCode["Gl"] = "GL";
  SupportedCountryCode["Gd"] = "GD";
  SupportedCountryCode["Gp"] = "GP";
  SupportedCountryCode["Gt"] = "GT";
  SupportedCountryCode["Gg"] = "GG";
  SupportedCountryCode["Gn"] = "GN";
  SupportedCountryCode["Gw"] = "GW";
  SupportedCountryCode["Gy"] = "GY";
  SupportedCountryCode["Ht"] = "HT";
  SupportedCountryCode["Hm"] = "HM";
  SupportedCountryCode["Va"] = "VA";
  SupportedCountryCode["Hn"] = "HN";
  SupportedCountryCode["Hk"] = "HK";
  SupportedCountryCode["Hu"] = "HU";
  SupportedCountryCode["Is"] = "IS";
  SupportedCountryCode["In"] = "IN";
  SupportedCountryCode["Id"] = "ID";
  SupportedCountryCode["Ir"] = "IR";
  SupportedCountryCode["Iq"] = "IQ";
  SupportedCountryCode["Ie"] = "IE";
  SupportedCountryCode["Im"] = "IM";
  SupportedCountryCode["Il"] = "IL";
  SupportedCountryCode["It"] = "IT";
  SupportedCountryCode["Jm"] = "JM";
  SupportedCountryCode["Jp"] = "JP";
  SupportedCountryCode["Je"] = "JE";
  SupportedCountryCode["Jo"] = "JO";
  SupportedCountryCode["Kz"] = "KZ";
  SupportedCountryCode["Ke"] = "KE";
  SupportedCountryCode["Ki"] = "KI";
  SupportedCountryCode["Kp"] = "KP";
  SupportedCountryCode["Xk"] = "XK";
  SupportedCountryCode["Kw"] = "KW";
  SupportedCountryCode["Kg"] = "KG";
  SupportedCountryCode["La"] = "LA";
  SupportedCountryCode["Lv"] = "LV";
  SupportedCountryCode["Lb"] = "LB";
  SupportedCountryCode["Ls"] = "LS";
  SupportedCountryCode["Lr"] = "LR";
  SupportedCountryCode["Ly"] = "LY";
  SupportedCountryCode["Li"] = "LI";
  SupportedCountryCode["Lt"] = "LT";
  SupportedCountryCode["Lu"] = "LU";
  SupportedCountryCode["Mo"] = "MO";
  SupportedCountryCode["Mg"] = "MG";
  SupportedCountryCode["Mw"] = "MW";
  SupportedCountryCode["My"] = "MY";
  SupportedCountryCode["Mv"] = "MV";
  SupportedCountryCode["Ml"] = "ML";
  SupportedCountryCode["Mt"] = "MT";
  SupportedCountryCode["Mq"] = "MQ";
  SupportedCountryCode["Mr"] = "MR";
  SupportedCountryCode["Mu"] = "MU";
  SupportedCountryCode["Yt"] = "YT";
  SupportedCountryCode["Mx"] = "MX";
  SupportedCountryCode["Md"] = "MD";
  SupportedCountryCode["Mc"] = "MC";
  SupportedCountryCode["Mn"] = "MN";
  SupportedCountryCode["Me"] = "ME";
  SupportedCountryCode["Ms"] = "MS";
  SupportedCountryCode["Ma"] = "MA";
  SupportedCountryCode["Mz"] = "MZ";
  SupportedCountryCode["Mm"] = "MM";
  SupportedCountryCode["Na"] = "NA";
  SupportedCountryCode["Nr"] = "NR";
  SupportedCountryCode["Np"] = "NP";
  SupportedCountryCode["Nl"] = "NL";
  SupportedCountryCode["An"] = "AN";
  SupportedCountryCode["Nc"] = "NC";
  SupportedCountryCode["Nz"] = "NZ";
  SupportedCountryCode["Ni"] = "NI";
  SupportedCountryCode["Ne"] = "NE";
  SupportedCountryCode["Ng"] = "NG";
  SupportedCountryCode["Nu"] = "NU";
  SupportedCountryCode["Nf"] = "NF";
  SupportedCountryCode["Mk"] = "MK";
  SupportedCountryCode["No"] = "NO";
  SupportedCountryCode["Om"] = "OM";
  SupportedCountryCode["Pk"] = "PK";
  SupportedCountryCode["Ps"] = "PS";
  SupportedCountryCode["Pa"] = "PA";
  SupportedCountryCode["Pg"] = "PG";
  SupportedCountryCode["Py"] = "PY";
  SupportedCountryCode["Pe"] = "PE";
  SupportedCountryCode["Ph"] = "PH";
  SupportedCountryCode["Pn"] = "PN";
  SupportedCountryCode["Pl"] = "PL";
  SupportedCountryCode["Pt"] = "PT";
  SupportedCountryCode["Qa"] = "QA";
  SupportedCountryCode["Cm"] = "CM";
  SupportedCountryCode["Re"] = "RE";
  SupportedCountryCode["Ro"] = "RO";
  SupportedCountryCode["Ru"] = "RU";
  SupportedCountryCode["Rw"] = "RW";
  SupportedCountryCode["Bl"] = "BL";
  SupportedCountryCode["Sh"] = "SH";
  SupportedCountryCode["Kn"] = "KN";
  SupportedCountryCode["Lc"] = "LC";
  SupportedCountryCode["Mf"] = "MF";
  SupportedCountryCode["Pm"] = "PM";
  SupportedCountryCode["Ws"] = "WS";
  SupportedCountryCode["Sm"] = "SM";
  SupportedCountryCode["St"] = "ST";
  SupportedCountryCode["Sa"] = "SA";
  SupportedCountryCode["Sn"] = "SN";
  SupportedCountryCode["Rs"] = "RS";
  SupportedCountryCode["Sc"] = "SC";
  SupportedCountryCode["Sl"] = "SL";
  SupportedCountryCode["Sg"] = "SG";
  SupportedCountryCode["Sx"] = "SX";
  SupportedCountryCode["Sk"] = "SK";
  SupportedCountryCode["Si"] = "SI";
  SupportedCountryCode["Sb"] = "SB";
  SupportedCountryCode["So"] = "SO";
  SupportedCountryCode["Za"] = "ZA";
  SupportedCountryCode["Gs"] = "GS";
  SupportedCountryCode["Kr"] = "KR";
  SupportedCountryCode["Ss"] = "SS";
  SupportedCountryCode["Es"] = "ES";
  SupportedCountryCode["Lk"] = "LK";
  SupportedCountryCode["Vc"] = "VC";
  SupportedCountryCode["Sd"] = "SD";
  SupportedCountryCode["Sr"] = "SR";
  SupportedCountryCode["Sj"] = "SJ";
  SupportedCountryCode["Se"] = "SE";
  SupportedCountryCode["Ch"] = "CH";
  SupportedCountryCode["Sy"] = "SY";
  SupportedCountryCode["Tw"] = "TW";
  SupportedCountryCode["Tj"] = "TJ";
  SupportedCountryCode["Tz"] = "TZ";
  SupportedCountryCode["Th"] = "TH";
  SupportedCountryCode["Tl"] = "TL";
  SupportedCountryCode["Tg"] = "TG";
  SupportedCountryCode["Tk"] = "TK";
  SupportedCountryCode["To"] = "TO";
  SupportedCountryCode["Tt"] = "TT";
  SupportedCountryCode["Ta"] = "TA";
  SupportedCountryCode["Tn"] = "TN";
  SupportedCountryCode["Tr"] = "TR";
  SupportedCountryCode["Tm"] = "TM";
  SupportedCountryCode["Tc"] = "TC";
  SupportedCountryCode["Tv"] = "TV";
  SupportedCountryCode["Ug"] = "UG";
  SupportedCountryCode["Ua"] = "UA";
  SupportedCountryCode["Ae"] = "AE";
  SupportedCountryCode["Gb"] = "GB";
  SupportedCountryCode["Us"] = "US";
  SupportedCountryCode["Um"] = "UM";
  SupportedCountryCode["Uy"] = "UY";
  SupportedCountryCode["Uz"] = "UZ";
  SupportedCountryCode["Vu"] = "VU";
  SupportedCountryCode["Ve"] = "VE";
  SupportedCountryCode["Vn"] = "VN";
  SupportedCountryCode["Vg"] = "VG";
  SupportedCountryCode["Wf"] = "WF";
  SupportedCountryCode["Eh"] = "EH";
  SupportedCountryCode["Ye"] = "YE";
  SupportedCountryCode["Zm"] = "ZM";
  SupportedCountryCode["Zw"] = "ZW";
  SupportedCountryCode["Zz"] = "ZZ";
})(exports.SupportedCountryCode || (exports.SupportedCountryCode = {}));

exports.DEFAULT_DISCOUNT_CODE_LENGTH = DEFAULT_DISCOUNT_CODE_LENGTH;
exports.DEFAULT_WEEK_START_DAY = DEFAULT_WEEK_START_DAY;
